tests/integration/cattletest/core/test_secrets.py

```diff
@@ -168,3 +169,76 @@ def test_secret_multi_host(secret_context, super_client):
     c = client.wait_success(c)
     assert c.state == 'running'
     assert c.hosts()[0].id == host2.id
+
+
+def test_service_secrets_fields(context, secret_context):
+    client = secret_context.client
+    env = _create_stack(client)
+
+    image_uuid = secret_context.image_uuid
+
+    sclient = secret_context.client
+    secret1 = sclient.create_secret(name=random_str(),
+                                    value='foo')
+    assert secret1.state == 'creating'
+    assert secret1.value == 'foo'
+    secret1 = sclient.wait_success(secret1)
+
+    secret1 = sclient.reload(secret1)
+    assert secret1.value is None
+
+    secret2 = sclient.create_secret(name=random_str(),
+                                    value='bar')
+    assert secret2.state == 'creating'
+    assert secret2.value == 'bar'
+    secret2 = sclient.wait_success(secret2)
+
+    secret2 = sclient.reload(secret2)
+    assert secret2.value is None
+
+    secrets = [
+        {
+           "name": "my_secret1",
```

Comment:
the name in here is different than secret.name. The secret.name is just a name of the secret we created in rancher api, while the name in launch config is what we want the filename of the secret stored on host to be